src/components/PBSCPasswordStrength.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+const greyColor = '#C4C4C4';
+const baseFullColor = ['#DC2619', '#FFA114', '#76BC21', '#349724', '#287C1A'];
+
+const strengthListDictionary = [
+  {
+    strength: 'EMPTY',
+    strengthLevel: 0,
+    suggestions: [],
+  },
+  {
+    strength: 'WEAK',
+    strengthLevel: 1,
+    suggestions: [],
+  },
+  {
+    strength: 'FAIR',
+    strengthLevel: 2,
+    suggestions: [],
+  },
+  {
+    strength: 'GOOD',
+    strengthLevel: 3,
+    suggestions: [],
+  },
+  {
+    strength: 'STRONG',
+    strengthLevel: 4,
+    suggestions: [],
+  },
+  {
+    strength: 'VERY_STRONG',
+    strengthLevel: 5,
+    suggestions: [],
+  },
+];
+
+const PBSCPasswordStrength = (props) => {
+  const {
+    width = '80%',
+    style,
+    isValid,
+    value,
+    passwordStrengthValidation,
+    strengthLabel = 'Password strength',
+    fractionLabel = 'of',
+  } = props;
+
+  const [passwordStrength, setPasswordStrength] = useState(
+    strengthListDictionary[0]
+  );
+
+  useEffect(() => {
+    if (value !== '') {
+      const findStrength = strengthListDictionary.find(
+        (list) => list.strength === passwordStrengthValidation.strength
+      );
+      if (findStrength) {
+        setPasswordStrength({
+          ...findStrength,
+          suggestions: passwordStrengthValidation.suggestions,
+        });
+        isValid(passwordStrengthValidation.valid);
+      }
+    } else {
+      setPasswordStrength(strengthListDictionary[0]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
+
+  const getSuggestionsText = () => {
```

Comment:
@BillyRan there seems to be a lint check warning here. Please remove getSuggestionsText if it is not used